lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,427 @@
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+import * as rtlUtils from '../../../__testutils__/rtlUtils';
+
+import userEvent from '@testing-library/user-event';
+
+import NxCombobox, {Props} from '../NxCombobox';
+
+describe('NxCombobox', function() {
+  const minimalProps: Props<string | number> = {
+        value: '',
+        onChange: () => {},
+        onSearch: () => {},
+        matches: []
+      },
+      rtlRender = rtlUtils.rtlRender(NxCombobox, minimalProps);
+
+  //mock scrollIntoView function using jest
+  window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+  it('renders a div with the nx-combobox class and the specified attributes', function() {
+    const { container } = rtlRender({ title: 'bar' }),
+        component = container.firstElementChild!;
+
+    expect(component.tagName).toBe('DIV');
+    expect(component).toHaveClass('nx-combobox');
+    expect(component).toHaveAttribute('title', 'bar');
+  });
+
+  it('adds the ref to the div', function() {
+    const ref = React.createRef<HTMLDivElement>(),
+        domNode = render(<NxCombobox { ...minimalProps } ref={ref} />).container.firstElementChild!;
+
+    expect(ref.current).toBeDefined();
+    expect(ref.current).toBe(domNode);
+  });
+
+  it('renders an input child with the combobox role', function() {
+    const { getByRole } = rtlRender(),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement.tagName).toBe('INPUT');
+    expect(inputElement).toHaveAttribute('role', 'combobox');
+  });
+
+  it('sets the value as the value of the input', function() {
+    const { getByRole } = rtlRender({ value: 'foo' }),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement).toHaveValue('foo');
+  });
+
+  it('sets aria-autocomplete on the input to list as default and to both if `autoComplete` prop is set to true',
+      function() {
+        const { getByRole, rerender } = rtlRender(),
+            inputElement = getByRole('combobox');
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'list');
+
+        rerender(<NxCombobox {...minimalProps} autoComplete={true} />);
+
+        expect(inputElement).toHaveAttribute('aria-autocomplete', 'both');
+      });
+
+  it('sets a completion string of the selected suggestion from matches when `autoComplete` prop is set to true',
+      async function() {
+        const onChange = jest.fn(),
+            { getByRole } = rtlRender({ autoComplete: true,
+              matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }],
+              onChange}),
+            inputElement = getByRole('combobox');
+
+        inputElement.focus();
+        await userEvent.type(inputElement, 'f');
+        expect(onChange).toBeCalledWith('Foo');
```

Comment:
I'm under the impression that the completion should only become the new value on blur. See https://www.w3.org/WAI/ARIA/apg/patterns/combobox/ :

> The automatically selected suggestion becomes the value of the combobox when the combobox loses focus unless the user chooses a different suggestion or changes the character string in the combobox.

Reply:
The automatically selected suggestion contains the portion that has not been typed by the user and those appears inline after the input cursor in the combobox within the selectionRange, when user typed new character, it is shown in the beginning of the selectionRange.

Reply:
Updated here fa25b519a02c928be04b85c5771de9636df9af2b.
As for the `backspace` key in autoComplete example, I tested it with `evt.preventDefault()`, when typed `ite`, it showed `Item 1` which is `autoSelectedVal`, then pressed `backspace` key, the value became `ite` which is the value, but I think I still need the value in the input to be updated, so it should be `Ite` after pressing `backspace` key, turns out using `onKeyUp` would have the input updated before event fired.